programs/whirlpool/src/instructions/v2/initialize_token_badge.rs

```diff
@@ -0,0 +1,44 @@
+use crate::state::*;
+use anchor_lang::prelude::*;
+use anchor_spl::token_interface::Mint;
+
+#[derive(Accounts)]
+pub struct InitializeTokenBadge<'info> {
+    pub whirlpools_config: Box<Account<'info, WhirlpoolsConfig>>,
+
+    #[account(has_one = whirlpools_config)]
+    pub whirlpools_config_extension: Box<Account<'info, WhirlpoolsConfigExtension>>,
+
+    #[account(address = whirlpools_config_extension.token_badge_authority)]
+    pub token_badge_authority: Signer<'info>,
+
+    pub token_mint: InterfaceAccount<'info, Mint>,
+
+    #[account(init,
+      payer = funder,
+      seeds = [
+        b"token_badge",
+        whirlpools_config.key().as_ref(),
+        token_mint.key().as_ref(),
+      ],
+      bump,
+      space = TokenBadge::LEN)]
+    pub token_badge: Account<'info, TokenBadge>,
+
+    #[account(mut)]
+    pub funder: Signer<'info>,
+
+    pub system_program: Program<'info, System>,
+}
+
+pub fn handler(
+    ctx: Context<InitializeTokenBadge>,
+) -> Result<()> {
+    Ok(ctx
+        .accounts
+        .token_badge
+        .initialize(
```

Comment:
Yes.

My first thought was to have a flag to allow for each extension. (such as allowPermanentDelegate, allowTransferHook ...)

But at some point I realized that in the end it was only a 0 or 1, whether or not to allow the pool to be created using that token per WhirlpoolsConfig.  Also, dangerous extension is only initializable at Mint initialization. So for example, adding PermanentDelegate extension after the initialization is impossible.

I reserved 128 bytes of expansion room just in case.